server/handler/entitlement.go

```diff
@@ -0,0 +1,33 @@
+package handler
+
+import (
+	"net/http"
+	"server/model"
+
+	"gorm.io/gorm"
+)
+
+type EntitlementsCount struct {
+	Count int64  `json:"count"`
+	Error string `json:"error"`
+}
+
+func GetNumOfAzureMarketplaceEntitlements(db *gorm.DB, w http.ResponseWriter, r *http.Request) {
+	var entitlements []model.AzureMarketplaceEntitlement
+	if tx := db.Model(&model.AzureMarketplaceEntitlement{}).Find(&entitlements); tx.Error != nil {
+		respondError(w, http.StatusInternalServerError, tx.Error.Error())
+		return
+	}
+	// iterate over the entitlements
+	resp := EntitlementsCount{}
+	for _, e := range entitlements {
+		// error message here is from RH API call, we should not make it into our own error, hence 200 response
```

Comment:
Maybe use a different 20x status code to signify OK but there is an error attached?

Reply:
The reason for using 200 here is that this API call was successful however the content from Red Hat API couldn't be fetched. In other words, the error message is there to help UI put up the right message.

Reply:
Yeah, understood that but was thinking of a way to distinguish between the two cases.  But just checking the content of error works too.